force-app/main/default/lwc/ldsCreateRecordWithRecordInput/ldsCreateRecordWithRecordInput.js

```diff
@@ -0,0 +1,115 @@
+import { LightningElement, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import {
+    createRecord,
+    getRecordCreateDefaults,
+    generateRecordInputForCreate
+} from 'lightning/uiRecordApi';
+import { reduceErrors } from 'c/ldsUtils';
+import ACCOUNT_OBJECT from '@salesforce/schema/Account';
+import NAME_FIELD from '@salesforce/schema/Account.Name';
+import AREANUMBER_FIELD from '@salesforce/schema/Account.AreaNumber__c';
+import EXPECTEDREVENUE_FIELD from '@salesforce/schema/Account.ExpectedRevenue__c';
+
+export default class LdsCreateRecordWithRecordInput extends LightningElement {
+    areaNumber;
+    areaNumberEditable;
+    error;
+    expectedRevenue;
+    expectedRevenueEditable;
+    name;
+    nameEditable;
+    recordInput;
+
+    @wire(getRecordCreateDefaults, { objectApiName: ACCOUNT_OBJECT })
+    loadAccountCreateDefaults({ data, error }) {
+        if (data) {
+            const accountObjectInfo =
+                data.objectInfos[ACCOUNT_OBJECT.objectApiName];
+            const recordDefaults = data.record;
+            // Creates an account record input with default field values
+            this.recordInput = generateRecordInputForCreate(
+                recordDefaults,
+                accountObjectInfo // ObjectInfo filters it to only fields that are createable
+            );
+            this.setState();
+        } else if (error) {
+            this.setState(error);
+        }
+    }
+
+    setState(error) {
+        if (error) {
+            this.nameEditable = false;
+            this.areaNumberEditable = false;
+            this.expectedRevenueEditable = false;
+            this.name = '';
+            this.areaNumber = '';
+            this.expectedRevenue = '';
+            this.error = error;
+        } else {
```

Comment:
In a real-life implementation, you'd do this for every field, most likely. But what if you only did the permission check on a single field? I need to start up an org to see how this would look, potentially, but having Name always editable, and only having one other field that you actually did the checks on would reduce this code footprint significantly.

Reply:
Change applied.